Feature/processing

sanju-shaji opened the pull request:
Added functionality to get path as user input and verify the path

sanju-shaji committed:
Added feature to take user input

sanju-shaji committed:
Added feature to take user input

sanju-shaji committed:
Added feature to take user input

sanju-shaji committed:
Added feature to take user input

sanju-shaji committed:
Added feature to take user input

sanju-shaji committed:
Added feature to take user input

SChandrakanth15 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
 
 //TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
 // click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
 public class Main {
```
Change the ClassName to "LogAggregatorTool".

sanju-shaji replied:
done

SambitKSahu99 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
 
+        File userFolder = new File(userfilePath);
+        String [] folderContents=userFolder.list();
+
```
Remove empty lines

sanju-shaji replied:
> Remove empty lines

ok.
anything else?

sanju-shaji replied:
done

SambitKSahu99 commented on src/main/java/org/example/Main.java:
Rename this fileName to LogAggregatorTool

sanju-shaji replied:
done

SambitKSahu99 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
+
+
+
+        if (!(userFolder.exists())) {
```
Try to avoid nested-if and find another ways

sanju-shaji replied:
resolved

SambitKSahu99 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
+        else {
+            System.out.println("Processing...");
+        }
+
```
Remove all empty and commented lines

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
 
-        System.out.println("Welcome To Java");
+        Scanner input = new Scanner(System.in);
+        System.out.println("Enter The File Path :");
```
Strings should be moved to constants.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
+        File userFolder = new File(userfilePath);
+        String [] folderContents=userFolder.list();
+
+
```
Remove unwanted empty lines.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
+
+
+        if (!(userFolder.exists())) {
+            System.out.println("Folder does not exists");
```
Strings should be moved to constants.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
+        if (!(userFolder.exists())) {
+            System.out.println("Folder does not exists");
+        } else if (folderContents.length == 0) {
+                System.out.println("Folder is Empty");
```
Strings should be moved to constants.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
+                System.out.println("Folder is Empty");
+            }
+        else {
+            System.out.println("Processing...");
```
Strings should be moved to constants.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
+        else {
+            System.out.println("Processing...");
+        }
+
```
Remove unwanted empty lines.

sanju-shaji replied:
done

ashutosh-2609 commented on .idea/gradle.xml:
please don't commit .idea folder to git this will create conflict for the collaborator when they tries to clone your repo.
You can edit in .gitignore file so that it will not consider it while commit

sanju-shaji replied:
ok

ashutosh-2609 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
     public static void main(String[] args) {
 
-        System.out.println("Welcome To Java");
+        Scanner input = new Scanner(System.in);
+        System.out.println("Enter The File Path :");
+        String userfilePath = input.nextLine();
 
+        File userFolder = new File(userfilePath);
```
Not a proper way of reading command line input

sanju-shaji replied:
modified the code to take input as cli

ashutosh-2609 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
-        System.out.println("Welcome To Java");
+        Scanner input = new Scanner(System.in);
+        System.out.println("Enter The File Path :");
+        String userfilePath = input.nextLine();
```
userfilePath  not in the camel case

sanju-shaji replied:
done

ashutosh-2609 commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,31 @@
+import java.io.File;
 import java.util.Scanner;
 
 //TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
```
These comments can be removed

sanju-shaji replied:
done

ashutosh-2609 commented on src/main/java/org/example/Main.java:
Proper formatting of file is recomended
shift + ctrl + alt + L enter ( for intelliJ IDE) this will format your entire file

sanju-shaji replied:
formatted

ashutosh-2609 reviewed (changes requested):
I would suggest learning about SOLID principles in JAVA and try implementing it
reference:-
https://youtu.be/oRA-xnqlB10?feature=shared

divya-sathyan commented on src/main/java/org/example/Main.java:
```diff
@@ -1,14 +1,24 @@
 
+        File f=new File(filePath);
+        if(!(f.exists()) && !(f.isDirectory())) {
+            System.out.println("Folder Not Found");
```
You have not done validation for .log files
Do that also

sanju-shaji replied:
added the validation

sanju-shaji committed:
implemented the review

sanju-shaji committed:
implemented the review

sanju-shaji committed:
implemented the review

divya-sathyan commented on src/main/java/org/example/Strings.java:
```diff
@@ -0,0 +1,16 @@
+
+
+
+
```
Remove all the empty lines

sanju-shaji replied:
done

vinodkadam22 commented on src/main/java/org/example/Strings.java:
```diff
@@ -0,0 +1,16 @@
+package org.example;
+
+public class Strings {
+
```
Please provide the proper class name, if it is constants class mention 
Eg. LogAggregatorToolConstants

sanju-shaji replied:
done

vinodkadam22 commented on src/main/java/org/example/Strings.java:
```diff
@@ -0,0 +1,16 @@
+package org.example;
+
```
Please provide the proper package name

sanju-shaji replied:
done

divya-sathyan commented on src/main/java/org/example/LogAggregatorTool.java:
```diff
@@ -0,0 +1,32 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.example.Strings.*;
```
Remove * import

sanju-shaji replied:
done

vinodkadam22 commented on src/main/java/org/example/Strings.java:
```diff
@@ -0,0 +1,16 @@
+
+public class Strings {
+
+
+    public static final String  invalidPath="Folder does not exists";
+
+    public static final String emptyFolder ="Folder is empty";
+    public static final  String processing ="Processing...";
+
+    public static final String extensionchecker=" is not a log file";
+
+
```
All the constants must be in alphabetic order
All the constants variables must be in upper case.

Eg. public static final String INVALID_FOLDER_PATH = "Folder does not exist, please provide the valid folder path"

sanju-shaji replied:
done

vinodkadam22 commented on src/main/java/org/example/Strings.java:
```diff
@@ -0,0 +1,16 @@
+
+
+
+
```
Please remove additional lines.

sanju-shaji replied:
done

divya-sathyan commented on src/main/java/org/example/LogAggregatorTool.java:
```diff
@@ -0,0 +1,32 @@
+            return;
+        }
+        for (String files : folderContents) {
+            if (!(files.endsWith(".log"))) {
```
Please check the invalid senario also

sanju-shaji replied:
done

divya-sathyan commented on src/main/java/org/example/LogAggregatorTool.java:
```diff
@@ -0,0 +1,32 @@
+        }
+        for (String files : folderContents) {
+            if (!(files.endsWith(".log"))) {
+                System.out.println(files + extensionchecker);
```
I want in count of valid and invalid files 
Example: There are 4 valid files present in the folder
there is 1 invalid file present in the folder
I dont wont it to print 4 times

sanju-shaji replied:
done.
also added the functionality to not show the count of invalid files if no invalid files are present

sanju-shaji committed:
revised the code based on comments and resolved the comments

sanju-shaji committed:
revised the code based on comments and resolved the comments

sanju-shaji committed:
revised the code based on comments and resolved the comments

sanju-shaji committed:
revised the code based on comments and resolved the comments

made the feature as a seperate class file

ashutosh-2609 commented on src/main/java/tool/log_aggregator/InputValidator.java:
```diff
@@ -0,0 +1,53 @@
+            return;
+        }
+        System.out.println(PROCESSING);
+        int logFileCount = 0;
+        int nonLogFilesCount = 0;
+        boolean noInvalidFlag = false;
+        long totalFilesCount = (Arrays.stream(folderContents).count());
+        for (String files : folderContents) {
+            if ((files.endsWith(".log"))) {
+                logFileCount += 1;
+            } else {
+                if (!(files.endsWith(".log"))) {
+                    nonLogFilesCount += 1;
+                    noInvalidFlag = true;
+                }
+            }
+        }
+        System.out.println(TOTAL_FILE_COUNT + totalFilesCount);
+        if (noInvalidFlag) {
+            System.out.println(nonLogFilesCount + SLASH + totalFilesCount + INVALID_LOG_FILES);
+        }
+        System.out.println(logFileCount + SLASH + totalFilesCount + VALID_LOG_FILES);
+    }
```
Here our only concern is to find the count of valid and invalid files
I would suggest to modify this logic and keep a generic message saying
"Total files found: 'a', Invalid files: 'b', Valid files: 'c'"

sanju-shaji replied:
resolved

ashutosh-2609 commented on src/main/java/tool/log_aggregator/LogAggregatorToolConstants.java:
```diff
@@ -0,0 +1,20 @@
+    public  static  final  String SLASH="\\";
+    public static final String TOTAL_FILE_COUNT="Total number file in the specified Folder : ";
+    public static final String VALID_LOG_FILES=" are valid files";
+
+
+
+
+
+
+
+
+
```
Remove all these empty lines and format the file

sanju-shaji replied:
done

ashutosh-2609 commented on .idea/gradle.xml:
Try to remove this in next commit

sanju-shaji replied:
ok

vinodkadam22 commented on src/main/java/tool/log_aggregator/InputValidator.java:
```diff
@@ -0,0 +1,53 @@
+        System.out.println(TOTAL_FILE_COUNT + totalFilesCount);
+        if (noInvalidFlag) {
+            System.out.println(nonLogFilesCount + SLASH + totalFilesCount + INVALID_LOG_FILES);
+        }
+        System.out.println(logFileCount + SLASH + totalFilesCount + VALID_LOG_FILES);
+    }
```
```suggestion
        }
        System.out.println(logFileCount + LogAggregatorToolConstants .SLASH + totalFilesCount + VALID_LOG_FILES);
    }
```

sanju-shaji replied:
done

sanju-shaji committed:
revised the code based on comments and resolved the comments

made the feature as a seperate class file

sanju-shaji committed:
revised the code based on comments and resolved the comments

made the feature as a seperate class file

ashutosh-2609 commented on src/main/java/tool/logaggregator/InputValidator.java:
```diff
@@ -0,0 +1,45 @@
+        int logFileCount = 0;
+        int nonLogFilesCount = 0;
+        boolean noInvalidFlag = false;
+        long totalFilesCount = (Arrays.stream(folderContents).count());
```
(Arrays.stream(folderContents).count())
do we really need to perform stream

sanju-shaji replied:
ive replaced it with folderContents.length

SChandrakanth15 commented on src/main/java/tool/log_aggregator/LogAggregatorTool.java:
```diff
@@ -0,0 +1,47 @@
+        String userFilePath = args[0];
+        File userFolder = new File(userFilePath);
+        String[] folderContents = userFolder.list();
+        if (!(userFolder.isDirectory())) {
```
Move all the validations to a separate class.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/tool/logaggregator/InputValidator.java:
```diff
@@ -0,0 +1,45 @@
+import java.util.Arrays;
+
+public class InputValidator {
+    String userFilePath;
```
Change the variable to "userFolderPath" to make more descriptive

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/tool/logaggregator/InputValidator.java:
```diff
@@ -0,0 +1,45 @@
+import java.io.File;
+import java.util.Arrays;
+
+public class InputValidator {
```
Make sure to use class level and method level comments.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/tool/logaggregator/InputValidator.java:
```diff
@@ -0,0 +1,45 @@
+        this.userFilePath = userFilePath;
+    }
+
+    public void validateInput() {
```
Use method level comments.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/tool/logaggregator/InputValidator.java:
```diff
@@ -0,0 +1,45 @@
+            if ((files.endsWith(".log"))) {
+                logFileCount += 1;
+            } else {
+                if (!(files.endsWith(".log"))) {
```
do we need to use this if condition here?

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/tool/logaggregator/InputValidator.java:
```diff
@@ -0,0 +1,45 @@
+    public void validateInput() {
+        File userFolder = new File(userFilePath);
+        String[] folderContents = userFolder.list();
+        if (!(userFolder.isDirectory())) {
```
move all validations into separate class, Extract the validation logic into its own method for reusability.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/tool/logaggregator/InputValidator.java:
```diff
@@ -0,0 +1,45 @@
+            System.out.println(LogAggregatorToolConstants.INVALID_PATH);
+            return;
+        }
+        if (folderContents.length == 0) {
```
move all validations into separate class, Extract the validation logic into its own method for reusability.

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/tool/logaggregator/InputValidator.java:
```diff
@@ -0,0 +1,45 @@
+package tool.logaggregator;
```
Move this class to sperate Validator package

sanju-shaji replied:
done

SChandrakanth15 commented on src/main/java/tool/logaggregator/LogAggregatorToolConstants.java:
```diff
@@ -0,0 +1,11 @@
+package tool.logaggregator;
```
Move this class to sperate constant package

sanju-shaji replied:
done

sanju-shaji committed:
revised the code based on comments and resolved the comments

made the feature as a seperate class file

sanju-shaji committed:
revised the code based on comments and resolved the comments

made the feature as a seperate class file

sanju-shaji committed:
revised the code based on comments and resolved the comments

made the feature as a seperate class file
made validations seperate in seperate methods

SChandrakanth15 commented on src/main/java/tool/logaggregator/validator/InputValidator.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    //  To check if the user input path is a valid directory
+    public boolean checkValidDirectory(File userFolder) {
```
change the method name checkValidDirectory to "isValidDirectory".

SChandrakanth15 commented on src/main/java/tool/logaggregator/validator/InputValidator.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    //    To check if the user input path is empty or not
+    public boolean checkEmpty(File userFolder) {
```
change the method name checkEmpty() to isFolderEmpty().

SChandrakanth15 commented on src/main/java/tool/logaggregator/validator/InputValidator.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    //  To display the number of valid and invalid file and display processing
+    public void processingFileDetails() {
```
change the method name processingFileDetails() to "processFileDetails()".

sanju-shaji replied:
ive changed to validate extension..which make more sense

SChandrakanth15 commented on src/main/java/tool/logaggregator/LogAggregatorTool.java:
```diff
@@ -0,0 +1,20 @@
+            if (validator.checkValidDirectory(userFolder) && validator.checkEmpty(userFolder)) {
+                validator.processingFileDetails();
+            }
+        } catch (Exception e) {
```
change "e" to exception.

SChandrakanth15 commented on src/main/java/tool/logaggregator/LogAggregatorTool.java:
```diff
@@ -0,0 +1,20 @@
+import java.io.File;
+
+public class LogAggregatorTool {
+    public static void main(String[] args) {
```
Keeps main() clean and move the other codes in to an other method

SambitKSahu99 commented on src/main/java/tool/logaggregator/validator/InputValidator.java:
```diff
@@ -0,0 +1,57 @@
+                logFileCount += 1;
+            }
+            if (!(files.endsWith(".log"))) {
+                nonLogFilesCount += 1;
```
Instead of using an extra flag, you can calculate the number of non-valid files by subtracting validLogFiles from totalFolderContents. Then, simply check if the result is 0 or not for validation.

sanju-shaji replied:
ok.implemented

vinodkadam22 commented on src/main/java/tool/logaggregator/LogAggregatorTool.java:
```diff
@@ -0,0 +1,20 @@
+    public static void main(String[] args) {
+        String userFolderPath = args[0];
+        File userFolder = new File(userFolderPath);
+        InputValidator validator = new InputValidator(userFolderPath);
```
```suggestion
        InputValidator inputValidator = new InputValidator(userFolderPath);
```

vinodkadam22 commented on src/main/java/tool/logaggregator/constants/LogAggregatorToolConstants.java:
```diff
@@ -0,0 +1,11 @@
+    public static final String SLASH = "\\";
+    public static final String TOTAL_FILE_COUNT = "Total number file in the specified Folder : ";
+    public static final String VALID_LOG_FILES = " are valid files";
+}
```
Add one line at last

vinodkadam22 commented on src/main/java/tool/logaggregator/validator/InputValidator.java:
Add proper documentation

vinodkadam22 commented on src/main/java/tool/logaggregator/validator/InputValidator.java:
```diff
@@ -0,0 +1,57 @@
+        if (noInvalidFlag) {
+            System.out.println(nonLogFilesCount + LogAggregatorToolConstants.SLASH + totalFilesCount + LogAggregatorToolConstants.INVALID_LOG_FILES);
+        }
+        System.out.println(logFileCount + LogAggregatorToolConstants.SLASH + totalFilesCount + LogAggregatorToolConstants.VALID_LOG_FILES);
+    }
+}
```
Add one line at last

sanju-shaji committed:
implemented all the required changes as comments

sanju-shaji committed:
implemented all the required changes as comments
